README.md

```diff
@@ -29,9 +29,194 @@ This library is written in modern JavaScript and is published in both CommonJS a
 ## Usage
 
 ```js
+import { generateKeyPair, getKeyPairFromMnemonic, getKeyPairFromSeed } from 'human-crypto-keys';
 
+const keyPair = await generateKeyPair('rsa');
+// => Generates a key pair with rsa encryption and provides information for recovery.
+
+const keyPairFromMnemonic = await getKeyPairFromMnemonic(keyPair.mnemonic, keyPair.algorithm);
+// => Generates the same key pair based on the mnemonic.
+
+const keyPairFromSeed = await getKeyPairFromSeed(keyPair.seed, keyPair.algorithm);
+// => Generates the same key pair based on the seed.
+
+```
+
+> ⚠️ human-crypto-keys depends on cryptographical modules that can increase the size of a project. Please keep in mind that, depending on your needs, dynamic imports might be suitable.
+
+## API
+
+### generateKeyPair(algorithm, options)
+
+Generates a key pair based on the specified algorithm.
+
+Returns an object with the following:
+```js
+{
+    algorithm,  // An object with the algorithm identifier and respective parameters that were used during generation.
+    mnemonic,   // The mnemonic used to create a seed for generation.
+    seed,       // The seed used for generation.
+    privateKey, // The generated private key composed in a specific format.
+    publicKey,  // The generated public key composed in a specific format. 
+}
+```
+
+#### Parameters
```

Comment:
Maybe change this to Arguments?

Reply:
As discussed, will remove this as it doesn't bring much more information.